sktime/forecasting/pytorchforecasting.py

```diff
@@ -0,0 +1,348 @@
+# copyright: sktime developers, BSD-3-Clause License (see LICENSE file)
+"""Interfaces to estimators from pytorch-forecasting."""
+import functools
+from typing import Any, Dict, List, Optional
+
+from sktime.forecasting.base.adapters._pytorchforecasting import (
+    _PytorchForecastingAdapter,
+)
+from sktime.utils.dependencies import _check_soft_dependencies
+
+__author__ = ["XinyuWu"]
+
+
+class PytorchForecastingTFT(_PytorchForecastingAdapter):
+    """pytorch-forecasting Temporal Fusion Transformer model.
+
+    Parameters
+    ----------
+    model_params :  Dict[str, Any] (default=None)
+        parameters to be passed to initialize the pytorch-forecasting TFT model [1]_
+        for example: {"lstm_layers": 3, "hidden_continuous_size": 10}
+    dataset_params : Dict[str, Any] (default=None)
+        parameters to initialize `TimeSeriesDataSet` [2]_ from `pandas.DataFrame`
+        max_prediction_length will be overwrite according to fh
+        time_idx, target, group_ids, time_varying_known_reals, time_varying_unknown_reals
+        will be infered from data, so you do not have to pass them
+    train_to_dataloader_params : Dict[str, Any] (default=None)
+        parameters to be passed for `TimeSeriesDataSet.to_dataloader()`
+        by default {"train": True}
+    validation_to_dataloader_params : Dict[str, Any] (default=None)
+        parameters to be passed for `TimeSeriesDataSet.to_dataloader()`
+        by default {"train": False}
+
+    References
+    ----------
+    .. [1] https://pytorch-forecasting.readthedocs.io/en/stable/api/pytorch_forecasting.models.temporal_fusion_transformer.TemporalFusionTransformer.html
+    .. [2] https://pytorch-forecasting.readthedocs.io/en/stable/api/pytorch_forecasting.data.timeseries.TimeSeriesDataSet.html
+    """  # noqa: E501
+
+    _tags = {
```

Comment:
I suppose some of these models are supporting probabilistic forecasting. Please check this. It is okay if you add the support later, but we should track this in this case via an issue

Reply:
I am not sure how probabilistic forecasting works, I will try to learn it. And @fkiraly suggested to implement it in the second PR.